app/oauth.go

```diff
@@ -722,28 +752,29 @@ func (a *App) AuthorizeOAuthUser(w http.ResponseWriter, r *http.Request, service
 	}
 
 	cookieValue := ""
-	if cookie, err := r.Cookie(COOKIE_OAUTH); err != nil {
+	cookie, cookieErr := r.Cookie(COOKIE_OAUTH)
+	if cookieErr != nil {
 		return nil, "", stateProps, model.NewAppError("AuthorizeOAuthUser", "api.user.authorize_oauth_user.invalid_state.app_error", nil, "", http.StatusBadRequest)
-	} else {
-		cookieValue = cookie.Value
 	}
 
+	cookieValue = cookie.Value
```

Comment:
You could drop this variable entirely and just use `cookie.Value` in it's place, since the refactor of the `if` means it is now in scope here.